### PR TITLE
test(integration/ci): Option to print debug messages from integration tests

### PR DESCRIPTION
**Pull request by petervdmeer**
**Feature**

This PR adds an option to the integration test framework to print messages to a terminal

## Summary
Integration tests running in CI were not very transparent, they either pass, fail or timeout, but it is not always clear why a test passes or fails. This PR adds an option to include debug-messages within integration tests to allow some more information about the internals of the test to be printed to the outside world.

## Screenshots
The output should look a bit like:
```
# Running test "Afterburner-flight - Simple depart land":
#     Departed from the planet
#     Start landing
#     Done landing
ok 2 Afterburner-flight - Simple depart land
```

## Usage examples
Add `debug` statements in integration tests, but please don't add them in long loops.

## Testing Done
Ran the integration test (in debug mode) and noticed the output messages appearing on the terminal.

## Wiki Update
I don't think the integration tests are documented on the wiki.

## Performance Impact
Test-code only. Should not impact the main game.

### CODE REVIEW

File: tests/integration/config/plugins/integration-tests/data/tests/tests_afterburn_flight.txt [hunk 1]
```diff
@@ -121,6 +122,8 @@ test "Afterburner-flight - Simple depart land"
 			"test: steps to wait" > 0
 		navigate
 			"travel destination" Ruin
+		debug "Start landing"
```
Comment (TheGiraffe3) [inline tests/integration/config/plugins/integration-tests/data/tests/tests_afterburn_flight.txt:125]:
```suggestion
		debug "Started landing"
```

**Comment by TheGiraffe3** [comment]
The integration tests are not documented on the wiki, though it might be good to do so.

File: tests/integration/config/plugins/integration-tests/data/tests/tests_afterburn_flight.txt (hunk 1 above)
Comment (warp-core) [reply tests/integration/config/plugins/integration-tests/data/tests/tests_afterburn_flight.txt:125]: The previous message is after the call to depart. In order to get there, we must have departed. This is before the call to land, and therefore past tense would be inappropriate.

**Comment by petervdmeer** [comment]
> Should DEBUG support substitutions in the messages? I could see that being useful in some cases.

Yes, supporting text replacements as is done for [writing conversations](https://github.com/endless-sky/endless-sky/wiki/WritingConversations) would be very elegant and useful.

Not sure if I get to it soon though (my Debian Linux install is having some issues with catch2). I would also be okay to keep substitutions for a follow-up PR.

**Pull request merged**